Build failure: lager

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

* Checkout recent master of nixpkgs e.g. b4fd7504eb4c022b45b01310539c2c92a5627310
* build `lager`
* cmake error should appear

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/322139025

### Relevant log output

```console
Running phase: unpackPhase
unpacking source archive /nix/store/bk9ygri9h8wxqxks4gzh6hxpbkrw81fi-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/share/doc/lager -DCMAKE_INSTALL_INFODIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0/lib -DCMAKE_STRIP=/nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/ranlib -DCMAKE_AR=/nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/i8hnkka6lkkhrg2p8bii5iy4dgcpaava-lager-0.1.0 -Dlager_BUILD_EXAMPLES=OFF
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/kbw2j1vag664b3sj3rjwz9v53cqx87sb-gcc-wrapper-15.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at CMakeLists.txt:40 (message):
  examples using the web-based debugger are disabled when examples are
  disabled


-- Could not find ccache
CMake Warning (dev) at CMakeLists.txt:72 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /nix/store/adv73gqmglkg48w5ql70s778ba6wpbs4-boost-1.89.0-dev/lib/cmake/Boost-1.89.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_system"
  (requested version 1.89.0) with any of the following names:

    boost_systemConfig.cmake
    boost_system-config.cmake

  Add the installation prefix of "boost_system" to CMAKE_PREFIX_PATH or set
  "boost_system_DIR" to a directory containing one of the above files.  If
  "boost_system" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /nix/store/adv73gqmglkg48w5ql70s778ba6wpbs4-boost-1.89.0-dev/lib/cmake/Boost-1.89.0/BoostConfig.cmake:262 (boost_find_component)
  /nix/store/1m0z4ixir10fg7ba7g131js04bxgv265-cmake-4.1.2/share/cmake-4.1/Modules/FindBoost.cmake:609 (find_package)
  CMakeLists.txt:72 (find_package)


-- Configuring incomplete, errors occurred!
```

### Additional context

I was trying to build on x86-64 on nixos (linux)
The build error is from the hydra build fail yesterday but my build error is the same, just harder to read.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.19.3, NixOS, 26.05 (Yarara), 26.05.git.0ccffdeb46f0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.94.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/julia/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/julia/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/julia/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/julia/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/julia/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/qjr1fm8hzppvjmmhk4slxxpa9iv3djfd-lix-2.94.0/share`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@nek0


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/la/lager/package.nix
+++ b/pkgs/by-name/la/lager/package.nix
@@ -2,11 +2,14 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   boost,
   cereal,
   immer,
   zug,
+  catch2,
+  qt5,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -20,6 +23,15 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-ssGBQu8ba798MSTtJeCBE3WQ7AFfvSGLhZ7WBYHEgfw=";
   };
 
+  patches = lib.optionals finalAttrs.finalPackage.doCheck [
+    # https://github.com/arximboldi/lager/pull/233
+    (fetchpatch {
+      name = "Stop-using-Boost-system.patch";
+      url = "https://github.com/arximboldi/lager/commit/0eb1d3d3a6057723c5b57b3e0ee3e41924ff419a.patch";
+      hash = "sha256-peGpuyuCznCDqYo+9zk1FytLV+a6Um8fvjLmrm7Y2CI=";
+    })
+  ];
+
   strictDeps = true;
 
   nativeBuildInputs = [ cmake ];
@@ -31,17 +43,26 @@ stdenv.mkDerivation (finalAttrs: {
     zug
   ];
 
+  checkInputs = [
+    catch2
+    qt5.qtdeclarative
+  ];
+
   cmakeFlags = [
     (lib.cmakeBool "lager_BUILD_DEBUGGER_EXAMPLES" false)
     (lib.cmakeBool "lager_BUILD_DOCS" false)
     (lib.cmakeBool "lager_BUILD_EXAMPLES" false)
-    (lib.cmakeBool "lager_BUILD_TESTS" false)
+    (lib.cmakeBool "lager_BUILD_TESTS" finalAttrs.finalPackage.doCheck)
   ];
 
   # remove BUILD file to avoid conflicts with the build directory
   preConfigure = ''
     rm BUILD
   '';
+
+  doCheck = true;
+
+  dontWrapQtApps = true;
 
   meta = {
     changelog = "https://github.com/arximboldi/lager/releases/tag/${finalAttrs.src.tag}";

--- a/pkgs/by-name/la/lager/package.nix
+++ b/pkgs/by-name/la/lager/package.nix
@@ -4,38 +4,49 @@
   fetchFromGitHub,
   cmake,
   boost,
+  cereal,
   immer,
   zug,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "lager";
-  version = "0.1.0";
+  version = "0.1.2";
 
   src = fetchFromGitHub {
     owner = "arximboldi";
     repo = "lager";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-KTHrVV/186l4klwlcfDwFsKVoOVqWCUPzHnIbWuatbg=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-ssGBQu8ba798MSTtJeCBE3WQ7AFfvSGLhZ7WBYHEgfw=";
   };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [ cmake ];
 
   buildInputs = [
     boost
+    cereal
     immer
     zug
   ];
-  nativeBuildInputs = [
-    cmake
-  ];
+
   cmakeFlags = [
-    "-Dlager_BUILD_EXAMPLES=OFF"
+    (lib.cmakeBool "lager_BUILD_DEBUGGER_EXAMPLES" false)
+    (lib.cmakeBool "lager_BUILD_DOCS" false)
+    (lib.cmakeBool "lager_BUILD_EXAMPLES" false)
+    (lib.cmakeBool "lager_BUILD_TESTS" false)
   ];
+
+  # remove BUILD file to avoid conflicts with the build directory
   preConfigure = ''
     rm BUILD
   '';
+
   meta = {
-    homepage = "https://github.com/arximboldi/lager";
+    changelog = "https://github.com/arximboldi/lager/releases/tag/${finalAttrs.src.tag}";
     description = "C++ library for value-oriented design using the unidirectional data-flow architecture — Redux for C++";
+    homepage = "https://sinusoid.es/lager/";
     license = lib.licenses.mit;
     maintainers = [ ];
   };

